pkg/util/pool/pool.go

```diff
@@ -176,3 +176,71 @@ func (b *BatchBytes) Get(sz int) ([]byte, error) {
 	*slab = (*slab)[:len(*slab)+sz]
 	return (*slab)[len(*slab)-sz : len(*slab) : len(*slab)], nil
 }
+
+// SlabPool is NOT concurrency safe.
+type SlabPool[T any] struct {
+	delegate Interface
+	slabSize int
+	slabs    []*[]T
```

Comment:
Sorry for the offtopic but `[]*[]T` is the state of the art of the expressiveness. 😂